site/viteUtils.tsx

```diff
@@ -0,0 +1,141 @@
+import React from "react"
+import findBaseDir from "../settings/findBaseDir.js"
+import fs from "fs-extra"
+import { ENV, BAKED_BASE_URL } from "../settings/serverSettings.js"
+import { POLYFILL_URL } from "./SiteConstants.js"
+import type { Manifest } from "vite"
+
+const VITE_DEV_URL = process.env.VITE_DEV_URL ?? "http://localhost:8090"
+
+// We ALWAYS load Google Fonts and polyfills.
+
+const googleFontsStyles = (
+    <link
+        key="google-fonts"
+        href="https://fonts.googleapis.com/css?family=Lato:300,400,400i,700,700i,900|Playfair+Display:400,600,700&display=swap"
+        rel="stylesheet"
+    />
+)
+
+const polyfillScript = <script key="polyfill" src={POLYFILL_URL} />
+
+interface Assets {
+    styles: JSX.Element[]
+    scripts: JSX.Element[]
+}
+
+// in dev: we need to load several vite core scripts and plugins; other than that we only need to load the entry point, and vite will take care of the rest.
+const devAssets = (entry: string): Assets => {
+    return {
+        styles: [googleFontsStyles],
+        scripts: [
+            polyfillScript,
+            <script
+                key="vite-react-preamble" // https://vitejs.dev/guide/backend-integration.html
+                type="module"
+                dangerouslySetInnerHTML={{
+                    __html: `import RefreshRuntime from '${VITE_DEV_URL}/@react-refresh'
+  RefreshRuntime.injectIntoGlobalHook(window)
+  window.$RefreshReg$ = () => {}
+  window.$RefreshSig$ = () => (type) => type
+  window.__vite_plugin_react_preamble_installed__ = true`,
+                }}
+            />,
+            <script
+                key="vite-plugin-checker"
+                type="module"
+                src={`${VITE_DEV_URL}/@vite-plugin-checker-runtime-entry`}
+            />,
+            <script
+                key="vite-client"
+                type="module"
+                src={`${VITE_DEV_URL}/@vite/client`}
+            />,
+            <script
+                key={entry}
+                type="module"
+                src={`${VITE_DEV_URL}/${entry}`}
+            />,
+        ],
+    }
+}
+
+// Goes through the manifest.json file that vite creates, finds all the assets that are required for the entry point,
+// and creates the appropriate <link> and <script> tags for them.
+const createTagsForManifestEntry = (
+    manifest: Manifest,
+    entry: string,
+    assetBaseUrl: string
+): Assets => {
```

Comment:
@danyx23 curious what you think about this method!

For reference, this is the manifest file that we're currently getting, which might as well be straight from hell:

<details>
<summary>manifest.json</summary>

```json
{
  "_common.mjs": {
    "css": [
      "assets/common.css"
    ],
    "file": "assets/common.mjs"
  },
  "adminSiteClient/admin.entry.css": {
    "file": "assets/admin.css",
    "src": "adminSiteClient/admin.entry.css"
  },
  "adminSiteClient/admin.entry.ts": {
    "css": [
      "assets/admin.css"
    ],
    "file": "assets/admin.mjs",
    "imports": [
      "_common.mjs"
    ],
    "isEntry": true,
    "src": "adminSiteClient/admin.entry.ts"
  },
  "faBook.css": {
    "file": "assets/common.css",
    "src": "faBook.css"
  },
  "site/owid.entry.css": {
    "file": "assets/owid.css",
    "src": "site/owid.entry.css"
  },
  "site/owid.entry.ts": {
    "css": [
      "assets/owid.css"
    ],
    "file": "assets/owid.mjs",
    "imports": [
      "_common.mjs"
    ],
    "isEntry": true,
    "src": "site/owid.entry.ts"
  }
}
```

</details>